service_container/service_subscribers_locators.rst

```diff
@@ -372,6 +372,218 @@ will share identical locators amongst all the services referencing them::
+
+If you want to retrieve a specific service within the injected service collector
+you can use the ``index_by`` and ``default_index_method`` options of the argument
+in combination with ``!tagged_locator`` to define an index.
+
+In the following example, all services tagged with ``app.handler`` are passed as
+first constructor argument to ``App\Handler\HandlerCollection``,
+but we can now access a specific injected service:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        services:
+            App\Handler\One:
+                tags:
+                    - { name: 'app.handler', key: 'handler_one' }
+
+            App\Handler\Two:
+                tags:
+                    - { name: 'app.handler', key: 'handler_two' }
+
+            App\HandlerCollection:
+                # inject all services tagged with app.handler as first argument
+                arguments: [!tagged_locator { tag: 'app.handler', index_by: 'key' }]
+
+    .. code-block:: xml
+
+        <!-- config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <service id="App\Handler\One">
+                    <tag name="app.handler" key="handler_one" />
+                </service>
+
+                <service id="App\Handler\Two">
+                    <tag name="app.handler" key="handler_two" />
+                </service>
+
+                <service id="App\HandlerCollection">
+                    <!-- inject all services tagged with app.handler as first argument -->
+                    <argument type="tagged_locator" tag="app.handler" index-by="key" />
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // config/services.php
+        use Symfony\Component\DependencyInjection\Argument\TaggedIteratorArgument;
+        use Symfony\Component\DependencyInjection\Argument\ServiceLocatorArgument;
+
+        $container->register(App\Handler\One::class)
+            ->addTag('app.handler', ['key' => 'handler_one']);
+
+        $container->register(App\Handler\Two::class)
+            ->addTag('app.handler', ['key' => 'handler_two']);
+
+        $container->register(App\Handler\HandlerCollection::class)
+            // inject all services tagged with app.handler as first argument
+            ->addArgument(new ServiceLocatorArgument(new TaggedIteratorArgument('app.handler', 'key')));
+
+After compilation the ``HandlerCollection`` to retrieve a specific service by it's ``key`` attribute
+from the service locator injected, we just have to do ``$serviceLocator->get('handler_two');`` to
+retrieve the ``handler_two`` handler :
+
+.. code-block:: php
```

Comment:
```suggestion

```

Reply:
why remove this one ?  my php code would not be correctly parse and show if we remove that line

Reply:
Thats why you should end the paragraph before with `::`

```
Here you can see my funky php highlighted::

    echo 'foo';
```

works

Reply:
oh :O 
Ok I did'nt know that possibility , thanks I'll make the changes